controllers/transaction.js

```diff
@@ -0,0 +1,86 @@
+const Transaction = require("../models/Transaction");
+const TransactionSlip = require("../models/TransactionSlip");
+
+// @desc    Update transaction
+// @route   PUT /api/v1/transactions/:id
+// @access  Private
+exports.updateTransaction = async (req, res, next) => {
+  try {
+    let transaction = await Transaction.findById(req.params.id);
+    // console.log(req.params.id);
+    //check transaction
+    if (!transaction) {
+      return res.status(404).json({
+        success: false,
+        message: `No transaction with the id of ${req.params.id}`,
+      });
+    }
+
+    // Status : [PENDING, COMPLETE, REJECTED, CANCELED]
+
+    //check status transaction is not COMPLETE and CANCELED
+    if (transaction.status === "COMPLETE" || transaction.status === "CANCELED") {
+      return res.status(404).json({
+        success: false,
+        message: `Transaction with the id of ${req.params.id}'s status is not available to update [Status: COMPLETE, Status: CANCELED]`,
+      });
+    }
+
+    //Check User or Admin
+
+    if (req.user.role === "admin") {
+      // ADMIN CASE: Admin can update when transaction was PENDING after user upload slip
+
+      //Check status of transaction
+      if (transaction.status !== "PENDING") {
+        return res.status(404).json({
+          success: false,
+          message: `Transaction with the id of ${req.params.id}'s status is not available to check [Status: REJECTED]`,
+        });
+      }
+
+    } else {
+      // USER CASE: When admin rejected transaction, USER will see transaction's status: PENDING 
+   
+      // Make sure user is the transaction owner
+      if (transaction.user.toString() !== req.user.id &&req.user.role !== "admin") {
```

Comment:
if user role is admin the code will never reach this line

Reply:
we check user role above this condition